core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -111,7 +115,11 @@ class RDDDataDistribution private[spark](
     val address: String,
     val memoryUsed: Long,
     val memoryRemaining: Long,
-    val diskUsed: Long)
+    val diskUsed: Long,
+    val onHeapMemoryUsed: Long,
+    val offHeapMemoryUsed: Long,
+    val onHeapMemoryRemaining: Long,
+    val offHeapMemoryRemaining: Long)
```

Comment:
I don't see any reason why the history server *couldn't* show this info.  Though we would be fine now by having them be defined without Option, I worry that will be a problem if we ever change the history server to show this info, since it won't always have it available.